iroha-cli/interactive/impl/interactive_transaction_cli.cpp

```diff
@@ -236,9 +238,12 @@ namespace iroha_cli {
       auto create_account = parser::parseValue<bool>(params[8]);
 
       if (not read_self.has_value() or not edit_self.has_value()
-          or not read_all.has_value() or not transfer_receive.has_value()
-          or not asset_create.has_value() or not create_domain.has_value()
-          or not roles.has_value() or not create_account.has_value()) {
+          or not read_all.has_value()
```

Comment:
maybe add a vararg fn for that?
```cpp
template<typename T>
bool has_no_value() {
  return not t.has_value();
}

template<typename T, typename... Args>
bool has_no_value(T t, Args... a) {
  return not t.has_value() or not has_no_value(a...);
}
```

Not sure but some c++14 feature may allow to write only one method

Reply:
To be more specific, the correct code is:
```C++
template <typename T> bool has_value(T &&t) { return t.has_value(); }

template <typename T, typename... Args> bool has_value(T &&t, Args &&... a) {
  return t.has_value() and has_value(std::forward<Args>(a)...);
}

if(not has_value(a, b, c, d, e)) { /* do something */ }
```

Reply:
Another way is actually just write:
```
// optionals can be safely casted to bool
if(not (a && b && c && d && e)) { /* do something */  }
```

Reply:
Seems that only [c++17 feature] :(

Reply:
Implicit cast to bool looks even better for me

Reply:
Nice talk you have. Any outcome?